near-sdk/src/store/vec/mod.rs

```diff
@@ -0,0 +1,453 @@
+        entry
+    }
+
+    /// Returns a mutable reference to the element at the `index` provided.
+    pub fn get_mut(&mut self, index: u32) -> Option<&mut T> {
+        let entry = self.get_mut_inner(index);
+        entry.value_mut().as_mut()
+    }
+
+    fn swap(&mut self, a: u32, b: u32) {
+        if a >= self.len() || b >= self.len() {
+            env::panic(ERR_INDEX_OUT_OF_BOUNDS);
+        }
+
+        if a == b {
+            // Short circuit if indices are the same, also guarantees uniqueness below
+            return;
+        }
+
+        let val_a = self.get_mut_inner(a).replace(None);
+        let val_b = self.get_mut_inner(b).replace(val_a);
+        self.get_mut_inner(a).replace(val_b);
+    }
+
+    /// Removes an element from the vector and returns it.
+    /// The removed element is replaced by the last element of the vector.
+    /// Does not preserve ordering, but is `O(1)`.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `index` is out of bounds.
+    pub fn swap_remove(&mut self, index: u32) -> T {
+        if self.is_empty() {
+            env::panic(ERR_INDEX_OUT_OF_BOUNDS);
+        }
+
+        self.swap(index, self.len() - 1);
+        expect_consistent_state(self.pop())
+    }
+
+    /// Removes the last element from a vector and returns it, or `None` if it is empty.
+    pub fn pop(&mut self) -> Option<T> {
+        if self.is_empty() {
+            None
+        } else {
+            let last_idx = self.len - 1;
+            self.len = last_idx;
+
+            // Replace current value with none, and return the existing value
+            self.get_mut_inner(last_idx).replace(None)
+        }
+    }
+
+    /// Inserts a element at `index`, returns an evicted element.
+    ///
+    /// # Panics
+    ///
+    /// If `index` is out of bounds.
+    pub fn replace(&mut self, index: u32, element: T) -> T {
+        self.get_mut_inner(index).replace(Some(element)).unwrap()
+    }
+
+    /// Returns an iterator over the vector. This iterator will lazily load any values iterated
+    /// over from storage.
+    pub fn iter(&self) -> Iter<'_, T> {
+        Iter::new(self)
+    }
+
+    /// Returns an iterator over the [`Vector`] that allows modifying each value. This iterator
+    /// will lazily load any values iterated over from storage.
+    pub fn iter_mut(&mut self) -> IterMut<'_, T> {
+        IterMut::new(self)
+    }
+}
+
+#[cfg(not(feature = "expensive-debug"))]
+impl<T> std::fmt::Debug for Vector<T>
+where
+    T: BorshSerialize + std::fmt::Debug,
+{
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("Vector").field("len", &self.len).field("prefix", &self.prefix).finish()
+    }
+}
+
+#[cfg(feature = "expensive-debug")]
+impl<T: std::fmt::Debug + BorshDeserialize> std::fmt::Debug for Vector<T>
+where
+    T: BorshSerialize,
+{
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self.iter().collect::<Vec<_>>().fmt(f)
```

Comment:
I am not  fan of calling `.fmt()` using method call syntax -- there's the same method on Debug and Display, and that might cause method resolution ambiguities. IIRC, the `failure` crate once broke a chunk of the ecosystem due to this issue. I don't think this is a problem in this particular case, but still, I suggest spelling this as 

```rust
fmd::Debug::fmt(&self.iter().collect::<Vec<_>>(), f)
```